### PR TITLE
test: added missing tests for the CronJob analyzer

### PR DESCRIPTION
**Pull request by VaibhavMalik4187**
## 📑 Description
- Fixed a small bug where pre-analysis was incorrectly appended to the results every time at the end of the for loop. This caused the result for a single cronjob failure to be appended multiple times in the final results.

- Added missing test cases to ensure proper testing of the CronJob analyzer. The addition of these missing test cases has increased the code coverage of this analyzer to over 96%.

Partially Addresses: https://github.com/k8sgpt-ai/k8sgpt/issues/889

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: pkg/analyzer/cronjob_test.go [hunk 1]
```diff
@@ -15,219 +15,165 @@ package analyzer
...
 
-	config := common.Analyzer{
-		Client: &kubernetes.Client{
-			Client: clientset,
-		},
-		Context:   context.Background(),
-		Namespace: "default",
-	}
+	startingDeadline := new(int64)
+	*startingDeadline = -7
```
Comment (JuHyung-Son) [inline pkg/analyzer/cronjob_test.go:35]: It would be nice to be able to tell if a variable is for a failure case or not just by its name.

**Review by JuHyung-Son — COMMENTED** [review]
nice catch.

just one question.
i guess comparing failureText can occur unexpected test fail in the future even code is right.
how do you think?

**Comment by VaibhavMalik4187** [comment]
> nice catch.
> 
> just one question. i guess comparing failureText can occur unexpected test fail in the future even code is right. how do you think?

Thanks,

Great question. I think you're right, there's a possibility of the failure text changing in the future and when that happens the test will break even though it shouldn't. 

I've compared failureText in tests for others analyzers as well. Looks like now would be a great time to update them as well. Thought?

File: pkg/analyzer/cronjob_test.go (hunk 1 above)
Comment (VaibhavMalik4187) [reply pkg/analyzer/cronjob_test.go:35]: Thanks for the suggestion! Cleaner code is always helpful for everyone.

**Comment by JuHyung-Son** [comment]
> > nice catch.
> > just one question. i guess comparing failureText can occur unexpected test fail in the future even code is right. how do you think?
> 
> Thanks,
> 
> Great question. I think you're right, there's a possibility of the failure text changing in the future and when that happens the test will break even though it shouldn't.
> 
> I've compared failureText in tests for others analyzers as well. Looks like now would be a great time to update them as well. Thought?

yes. it needs to be. but not urgent. feel free to do that.

**Pull request merged**